### PR TITLE
template useability docs changes

### PR DESCRIPTION
**Pull request by AlecBoutin**
- tweaks for revised template
- move breaking changes to bottom, clarify instance name usage, reorg DET options

# Amplitude Developer Docs PR

## Description

Describe your changes. 

## Deadline

When do these changes need to be live on the site?

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ ] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [ ] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [ ] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/sources/google-tag-manager-client.md [hunk 1]
```diff
@@ -369,3 +339,33 @@ If you hardcode the value in your tag, the input will be forced into a string ty
...
+Traffic from one subdomain to another (ie analytics.amplitude.com to experiment.amplitude.com) is not tracked by default. If you want to exclude the attribution tracking on `location.hostname`, but not other subdomains, add the value of `location.hostname` in the exclude referral section. See full details in table. 
+
+???Breaking change "Attribution Tracking"
+    | <div class="big-column">Before</div>  | Current |
+    | --- | --- | --- | --- |
+    | Track attribution of all subdomains. | Excludes all subdomains of the same root domain as referrer | 
+
+### User agent parser
+
+The new template changes the way to parse the device related info which might effect the value of `event.os_name`, `event.os_version`, `event.device_model`, `event.device_manufacturer` and related properties. If you want to continue using the older way to parse user agent, check `Enable client side user agent enrichment `. See full details in table.
```
Comment (yuhao900914) [inline docs/data/sources/google-tag-manager-client.md:365]: Should we map the check box `Enable client side user agent enrichment` to the new version?

Comment (yuhao900914) [reply docs/data/sources/google-tag-manager-client.md:365]: `(Legacy) Enable client side user agent enrichment`

File: docs/data/sources/google-tag-manager-client.md [hunk 2]
```diff
@@ -160,33 +151,12 @@ If the user ID is already available you can:
 
 ##### Enable client side user agent enrichment
 
-Starting with version xxxxx, as of Aug, 2023, we have upgraded this template to use the Amplitude Browser SDK 2.0 for data collection. In Amplitude Browser SDK 2.0, we have deprecated client-side user agent parsing in favor of server-side user agent parsing. [More details](https://github.com/amplitude/Amplitude-TypeScript/tree/v1.x/packages/plugin-user-agent-enrichment-browser).
+Starting with version 240a7 in Aug, 2023, we have upgraded this template to use the Amplitude Browser SDK 2.0 for data collection. In Amplitude Browser SDK 2.0, we have deprecated client-side user agent parsing in favor of server-side user agent parsing. [More details](https://github.com/amplitude/Amplitude-TypeScript/tree/v1.x/packages/plugin-user-agent-enrichment-browser).
 
 To avoid breaking changes in chart analytics, we provide the option to enable client-side parsing by simply checking the check box. You can also choose which user agent field (OS name, OS version, device manufacture, and device model) you want to enrich by checking the checkbox accordingly.
```
Comment (yuhao900914) [inline docs/data/sources/google-tag-manager-client.md:156]: (Legacy) Enable client side user agent enrichment check box

File: docs/data/sources/google-tag-manager-client.md [hunk 3]
```diff
@@ -117,6 +88,26 @@ A tag type allow you to specify what kind of action or event should be tracked i
 
 Cookies are generated at the initialization stage. For more information on managing cookies, please refer to our [cookie management details](../../sdks/typescript-browser/#cookie-management). It is recommended to initialize after obtaining cookie consent. As we support deferred initialization, any event tracking that takes place before the init command will still be captured and won't be lost.
 
+##### Default Event Tracking
```
Comment (yuhao900914) [inline docs/data/sources/google-tag-manager-client.md:91]: Track default Amplitude events

**Comment by yuhao900914** [comment]
It will be better if we also change the types accordingly to the new name.
Screen Shot 2023-08-16 at 5 22 18 PM

**Comment by AlecBoutin** [comment]
I'm hesitant to change those because it would require going back into the template and updating the doc links for all of the tag types. Imo the names in parentheses are sufficient.

**Review by yuhao900914 — APPROVED** [review]

File: docs/data/sources/google-tag-manager-client.md [hunk 4]
```diff
@@ -369,3 +339,33 @@ If you hardcode the value in your tag, the input will be forced into a string ty
 This video tutorial walks through the implementation basics. 
 
 <script src="https://fast.wistia.com/embed/medias/n337njhoot.jsonp" async></script><script src="https://fast.wistia.com/assets/external/E-v1.js" async></script><div class="wistia_responsive_padding" style="padding:56.25% 0 0 0;position:relative;"><div class="wistia_responsive_wrapper" style="height:100%;left:0;position:absolute;top:0;width:100%;"><div class="wistia_embed wistia_async_n337njhoot videoFoam=true" style="height:100%;position:relative;width:100%"><div class="wistia_swatch" style="height:100%;left:0;opacity:0;overflow:hidden;position:absolute;top:0;transition:opacity 200ms;width:100%;"><img src="https://fast.wistia.com/embed/medias/n337njhoot/swatch" style="filter:blur(5px);height:100%;object-fit:contain;width:100%;" alt="" aria-hidden="true" onload="this.parentNode.style.opacity=1;" /></div></div></div></div>
+
+## Breaking changes checklist
```
Comment (justin-fiedler) [inline docs/data/sources/google-tag-manager-client.md:343]: I think we should link to this from the [warning above](https://github.com/Amplitude-Developer-Docs/amplitude-dev-center/pull/904/files#diff-86171b1a11fcdb2c74f5676859fff852c835ef5693ca679ea5b507338d2a061bR12) to make it easy for folks to find.

**Review by justin-fiedler — APPROVED** [review]

File: docs/data/sources/google-tag-manager-client.md (hunk 4 above)
Comment (yuhao900914) [reply docs/data/sources/google-tag-manager-client.md:343]: I will add this in other pr.

**Pull request merged**